types/react-native/index.d.ts

```diff
@@ -3518,7 +3518,7 @@ export interface FlatListProperties<ItemT> {
      * ```
      * Provides additional metadata like `index` if you need it.
      */
-    renderItem: (info: ItemT) => React.ReactElement<any> | null
+    renderItem: (info: ItemT, index: number) => React.ReactElement<any> | null
```

Comment:
This actually seems wrong altogether, there should be 1 parameter, which is `info`, which is an object that holds `item` and `index`, just like is done in the [`SectionListData` interface](https://github.com/sverrejoh/DefinitelyTyped/blob/f0231257f101cec786c450811ec8b5cf90b6e3f4/types/react-native/index.d.ts#L3578).

Having said that, both [`FlatList`](https://facebook.github.io/react-native/docs/flatlist.html#renderitem) and [`SectionList`](https://facebook.github.io/react-native/docs/sectionlist.html) wrap [`VirtualizedList`](https://facebook.github.io/react-native/docs/virtualizedlist.html), which is the one that _actually_ has these shared props, so I think it’s best if you could add an interface for `VirtualizedList` and extend both `SectionList` and `FlatList` from it.